modin/core/execution/python/implementations/pandas_on_python/partitioning/partition.py

```diff
@@ -41,17 +41,30 @@ class PandasOnPythonDataframePartition(PandasDataframePartition):
     subclasses. There is no logic for updating in-place.
     """
 
-    def __init__(self, data, length=None, width=None, call_queue=None):
-        self.data = data
+    def __init__(self, physical_data, length=None, width=None, call_queue=None):
+        self._physical_data = physical_data
         if call_queue is None:
             call_queue = []
         self.call_queue = call_queue
         self._length_cache = length
         self._width_cache = width
 
+    @property
+    def physical_data(self):
+        """
+        Flush the `call_queue` and get the physical data wrapped by this partition.
+
+        Returns
+        -------
+        pandas.DataFrame
+            Copy of pandas DataFrame wrapped by this partition.
+        """
+        self.drain_call_queue()
```

Comment:
I don't think we want to drain the call queue every time when accessing the data. We likely want to drain the call queue in a worker.

Reply:
I second this. It would cause a lot of draining, is this what we really want?

Reply:
In case of accessing to physical data holded by partition the getter property should return actual data after all deffered operations. For example, in a case of creating the new partition from another one using physical data the correct state of source-partition will be lost in case not calling the call queue.

In the current code, before all accesses to physical data using `oid/future` `drain_call_queue` is called for the each partition.
If `call_queue` is empty, the call `drain_call_queue` doesn't include any additional computations.

> We likely want to drain the call queue in a worker.

@YarShev, what do you mean here?

Reply:
> > We likely want to drain the call queue in a worker.
> 
> @YarShev, what do you mean here?

Imagine a case when the partition has the call queue not drained and we want to pass it as an input in a remote call. It might be beneficial (I think it is rather beneficial) to pass the call queue along with the partition and drain the call queue of the partition in the remote call rather than perform that before.

I understand your concern on the actual physical data held by a partition. We have two options here.
1. Not drain the call queue in the getter and hand on this responsibility to the partition manager that handles partitions.
2. Drain the call queue in the getter and hand on the responsibility of accessing private `_data` field and draining the call queue to the partition manager in case we want to achieve the behavior which I said about above.

Do you think which option makes much more sense?

Reply:
> Imagine a case when the partition has the call queue not drained and we want to pass it as an input in a remote call. It might be beneficial (I think it is rather beneficial) to pass the call queue along with the partition and drain the call queue of the partition in the remote call rather than perform that before.

Do you mean case when we want pass some other `partition1` together with their `partiiton1.call_queue` in a remote call of `partition0`? Something like that:

```python
# partition manager impl
def some_operation(cls, foo, left_parts, other_parts):
   ...
   def func(lpart, rpart, rpart_call_queue, *args,  **kwargs):
      actual_rpart = rpart_call_queue(rpart) # drain call queue of other partition in remote call
      return foo(lpart, rpart)

   [lpart.apply(func, rpart._data, rpart.call_queue) for lpart, rpart in zip(left_parts, other_parts)] #--> We have access to private `partition._data`.

```
If you mean that I've described above, that it will be better the 3rd option, in my opinion:

3. Direct access to `_physical_data/_data` (naming discussion is in other thread) must happen only in partition class to resolve issues described in #4530. But instead of property for accessing to `_physical_data/_data` need to use getter func with parameter `is_drain_call_queue` with default value `True`. This approach allows you to transfer responsibility of draining call queue to partition manager directly with expected behavior for such cases. Also you will always have the actual physical data from default call `actual_data = prtition.get_data()`. 

```python
# partition class

def get_data(self, is_drain_call_queue=True):
   if is_drain_call_queue:
      self.drain_call_queue
   return self._data
   
```
What do you think @YarShev?

cc @vnlitvinov, @mvashishtha @modin-project/modin-core

Reply:
Yes, I mean that that you linked above. I think this option makes sense too with the getter named just `data()`. However, other question has arisen in my head. If we go with this option, do we need `drain_call_queue()` to be a public method? It seems to me we should make it private in that case.

Reply:
Also, there might be a confusion between `data()` and `get()` methods.

Reply:
I would probably narrow the scope of the issue just to standardizing access to physical data of partitions (not introducing the getter) since we also use internal members such as `._width_cache` and `._length_cache` outside of a partition class. Accessing the fields that might be corrupted is rather a broader problem and should probably be resolved as part of a separate issue.

Reply:
> Yes, I mean that that you linked above. I think this option makes sense too with the getter named just `data()`. However, other question has arisen in my head. If we go with this option, do we need `drain_call_queue()` to be a public method? It seems to me we should make it private in that case.

`data()` is also makes sense to me though I prefer to use verbs in functions/methods to highlight "action".

> Also, there might be a confusion between data() and get() methods.

Probably, it will be need to change name of `get` to something other with additional highlighting of data materialization.

> I would probably narrow the scope of the issue just to standardizing access to physical data of partitions (not introducing the getter) since we also use internal members such as ._width_cache and ._length_cache outside of a partition class. Accessing the fields that might be corrupted is rather a broader problem and should probably be resolved as part of a separate issue.

Agree, I'll change the scope of this PR to cover only 1st problem described in #4530. It will unblock PR #4391.
I'm not sure that examples of usage `._width_cache`/`._length_cache` outside of partition class is a good approach. Possible, for these properties it will be need also check/fix publicity.

Reply:
PR scope was changed to cover only 1st problem described in #4530 (renaming `oid/future/data -> _data`). The public getter will be added in an other PR as resolution of 2nd and 3rd problems from #4530.